src/server/applications/controller.js

```diff
@@ -29,6 +29,132 @@ const getCaseById = async (caseId) => {
   }
 }
 
+const updateStageAsync = async (caseId, nextStage) => {
+  try {
+    const backendUrl = config.get('fg_cw_backend_url')
+    const response = await fetch(
+      `${backendUrl.toString()}/cases/${caseId}/stage`,
+      {
+        method: 'POST',
+        data: {
+          nextStage
+        }
+      }
+    )
+    const data = await response.json()
+    return data
+  } catch {
+    return null
+  }
+}
+
+const showApplication = async (request, h) => {
+  const caseId = request.params.id
+  const selectedCase = await getCaseById(caseId)
+
+  if (!selectedCase) {
+    return h.response('Case not found').code(404)
+  }
+  const { workflowCode } = selectedCase
+
+  if (!workflowCode) {
+    return h.response('Workflow not found').code(404)
+  }
+  const workflow = await getWorkflowByCode(workflowCode)
+
+  // Add titles from workflow stages to selectedCase stages
+  selectedCase.stages = selectedCase.stages.map((stage) => {
+    const workflowStage = workflow.stages.find((ws) => ws.id === stage.id)
+
+    // Add title from workflow to the stage
+    const updatedStage = {
+      ...stage,
+      title: workflowStage?.title
```

Comment:
The contract enforces presence 

```suggestion
      title: workflowStage.title
```

Reply:
not my changes